docs/source/plug-in.md

```diff
@@ -0,0 +1,26 @@
+## Other modes
+
+The `KernelGatewayApp.api` can be set to the name of any public module supplying a personality. This allows for suporting alternate kernel communications mechanisms.
+
+### Factory function
+`create_personality(*args, **kwargs)`
+
+The only function required in the module, it is passed a named argument `parent` whose value is the kernel gateway application. The function must return an object implementing the following lifecycle and delegated functions:
+
+### Lifecycle functions
```

Comment:
After the template, relative links to the two files implementing the jupyter-websocket and notebook-http modes / personalities would be helpful.